ksqldb-engine/src/main/java/io/confluent/ksql/engine/PullQueryExecutionUtil.java

```diff
@@ -51,7 +51,8 @@ static PersistentQueryMetadata findMaterializingQuery(
     final Set<QueryId> queries = engineContext.getQueryRegistry().getQueriesWithSink(sourceName);
 
     if (source.getDataSourceType() != DataSourceType.KTABLE) {
-      throw new KsqlException("Pull queries are not supported on streams."
+      throw new KsqlException("Unexpected data source type for table pull query: "
+          + source.getDataSourceType() + " "
```

Comment:
This is now just serving as a last-minute check on the source type. As far as we know, though, all code paths leading here will only send us table queries now, so it makes sense to reword the error.